cpu/atmega_common/include/periph_cpu_common.h

```diff
@@ -61,7 +63,11 @@ typedef uint8_t gpio_t;
  *
  * Must be identical to the address of `PINA` provided by avr/io.h
  */
+#ifdef CPU_ATMEGA8
```

Comment:
This difference is not exclusive of Mega8.

My suggestion is check by

```diff
+#if (PORTB == _SFR_IO8(0x18))
+#define GPIO_PORT_DESCENDENT
+#endif

+#ifdef GPIO_PORT_DESCENDENT
+#define ATMEGA_GPIO_BASE_A     (0x39)
+#else
#define ATMEGA_GPIO_BASE_A      (0x20)
+#endif

...

+#ifdef GPIO_PORT_DESCENDENT
+    uintptr_t res = base_addr - port_num * sizeof(atmega_gpio_port_t);
+#else
    uintptr_t res = base_addr + port_num * sizeof(atmega_gpio_port_t);
+#endif
```

Reply:
```
cpu/atmega_common/include/periph_cpu_common.h:67:6: error: operator '*' has no left operand
 #if (PORTB == _SFR_IO8(0x18))
      ^
```

Reply:
It seems that pre-processor not handles same way c compiler. I did a research to understand GPIO address ASC/DESC but it is not possible to find a way to do it. In this case, each processor should define it.

I let to build experts define what could be best solution here since there is no trivial solution.

Just to mention but I think this may not be accepted as a solution. 

```
#undef _MMIO_BYTE
#define _MMIO_BYTE(mem_addr) (mem_addr)
#if (PORTB == 0x38)
#define GPIO_PORT_DESCENDENT
#endif
#undef _MMIO_BYTE
#define _MMIO_BYTE(mem_addr) (*(volatile uint8_t *)(mem_addr))
```